win32ss/user/conime/conime.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * PROJECT:     ReactOS Console IME
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Console IME
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#pragma once
+
+#define WIN32_NO_STATUS
+#include <windef.h>
+#include <winbase.h>
+#include <wingdi.h>
+#include <winuser.h>
+#include <winnls.h>
+#include <winreg.h>
+#include <kernel32_undoc.h>
+#include <undocuser.h>
+#include <imm.h>
+#include <immdev.h>
+#include <imm32_undoc.h>
+
+#define NTOS_MODE_USER
+#include <ndk/umtypes.h>
+#include <ndk/psfuncs.h>
+#include <ndk/sefuncs.h>
+#include <ndk/obfuncs.h>
+#include <ndk/rtlfuncs.h>
+
+#include <stdlib.h>
+#include <strsafe.h>
+
+/* PSEH for SEH Support */
+#include <pseh/pseh2.h>
+
+#include "resource.h"
+
+typedef struct tagKLINFO
+{
+    HKL hKL;
+    DWORD dwImeState;
+} KLINFO, *PKLINFO;
+
+typedef struct tagLOCAL1
+{
+    DWORD cbSize;
+    DWORD Unknown1;
+} LOCAL1, *PLOCAL1;
+
+typedef struct tagENTRY
+{
+    BOOL bEnabled;
+    HWND hwndConsole;
+    DWORD unknown0[1];
+    HKL hKL;
+    HIMC hOldIMC;
+    HIMC hIMC;
+    BOOL bOpened;
+    DWORD dwConversion;
+    DWORD dwSentence;
+    WORD unknown1;
```

Comment:
Why? Is this ENTRY structure communicated with other external components (the console, win32k, etc.)?
If not, why do you have "unknown" or other such members in that entry and others?

Reply:
No. This `ENTRY` structure is for internal use.